### ISSUE TITLE
./run started but `ios_video_stream` keep ending

### ISSUE THREAD

**Issue by github-june** [DeviceFarmer/stf_ios_support#75]
I executed './run', but it failed with following error.
`ios_video_stream`  keep ending:

![001](https://user-images.githubusercontent.com/16497823/95730342-b7102c80-0cb0-11eb-919b-7a4f42ad6e6d.png)

----

When open Stf browser, I can control iphone by mouse, but see nothing about phone's screen-video, such as:

![002](https://user-images.githubusercontent.com/16497823/95730672-2a19a300-0cb1-11eb-99a3-e9c3180031df.png)

----
This is my config.json : 

![003](https://user-images.githubusercontent.com/16497823/95731036-972d3880-0cb1-11eb-9596-d730db389f28.png)

----

I use 'docker-compose up' to start STF server.

Thanks so much for someone who can answer my doubts. Thanks!

**Comment by mbilbiesi** [comment]
Same behavior as in https://github.com/DeviceFarmer/stf_ios_support/issues/71 ; based on the owner of this repo; there will be a fix for this case !

**Comment by nanoscopic** [comment]
The cases I've seen where ios_video_stream refuses to start are due to there being old leftover copies of it still running. Recent changes fixed the removal of those so this should not be occurring on current fresh clone / build etc.

You can look at `./view_log -proc ios_video_stream` to see if it is printing out any error.

Without additional information on where ios_video_stream crashed I have no way to address this issue beyond what I've seen happen myself.

I've also seen ios_video_stream refuse to start due to the network interface not being specified. Recent changes default the network interface to the one that is setup to route by default so that problem should be resolved now also.

When this ticket was opened that was still an issue so it was necessary for the network interface to be specified in config.json to get ios_video_stream to start up correctly. That is no longer the case.

TLDR: I don't believe this is happening any longer with latest updates.

**Comment by Jitu1888** [comment]
@famgy How to start ios_video_stream and ios_video_pull ?

**Comment by Jitu1888** [comment]
![Uploading image.png…]()

**Comment by Jitu1888** [comment]
I m not seeing any logs related to ios_video_stream and ios_video_pull ? @famgy

**Comment by Jitu1888** [comment]
proc=stf_device_ios type=proc_end uuid="***402E"
INFO[0205] Process start - stf_device_ios                binary=/usr/local/opt/node@14/bin/node clickHeight=896 clickScale=1000 clickWidth=414 client_ip=192.168.1.11 device_name="Jitu iPhone Max" frame_server="ws://192.168.1.11:8000/echo" node_port=9240 proc=stf_device_ios server_host=127.0.0.1 server_ip=127.0.0.1 stream_height=0 stream_width=0 type=proc_start uuid="***402E" video_port=8000 vnc_scale=2
WARN[0206] Process end - stf_device_ios                  proc=stf_device_ios type=proc_end uuid="***402E"
INFO[0216] Process start - stf_device_ios                binary=/usr/local/opt/node@14/bin/node clickHeight=896 clickScale=1000 clickWidth=414 client_ip=192.168.1.11 device_name="Jitu iPhone Max" frame_server="ws://192.168.1.11:8000/echo" node_port=9240 proc=stf_device_ios server_host=127.0.0.1 server_ip=127.0.0.1 stream_height=0 stream_width=0 type=proc_start uuid="***402E" video_port=8000 vnc_scale=2
WARN[0216] Process end - stf_device_ios                  proc=stf_device_ios type=proc_end uuid="***402E"
INFO[0226] Process start - stf_device_ios                binary=/usr/local/opt/node@14/bin/node clickHeight=896 clickScale=1000 clickWidth=414 client_ip=192.168.1.11 device_name="Jitu iPhone Max" frame_server="ws://192.168.1.11:8000/echo" node_port=9240 proc=stf_device_ios server_host=127.0.0.1 server_ip=127.0.0.1 stream_height=0 stream_width=0 type=proc_start uuid="***402E" video_port=8000 vnc_scale=2
WARN[0227] Process end - stf_device_ios                  proc=stf_device_ios type=proc_end uuid="***402E"
INFO[0237] Process start - stf_device_ios                binary=/usr/local/opt/node@14/bin/node clickHeight=896 clickScale=1000 clickWidth=414 client_ip=192.168.1.11 device_name="Jitu iPhone Max" frame_server="ws://192.168.1.11:8000/echo" node_port=9240 proc=stf_device_ios server_host=127.0.0.1 server_ip=127.0.0.1 stream_height=0 stream_width=0 type=proc_start uuid="***402E" video_port=8000 vnc_scale=2
WARN[0237] Process end - stf_device_ios                  proc=stf_device_ios type=proc_end uuid="***402E"
INFO[0247] Process start - stf_device_ios                binary=/usr/local/opt/node@14/bin/node clickHeight=896 clickScale=1000 clickWidth=414 client_ip=192.168.1.11 device_name="Jitu iPhone Max" frame_server="ws://192.168.1.11:8000/echo" node_port=9240 proc=stf_device_ios server_host=127.0.0.1 server_ip=127.0.0.1 stream_height=0 stream_width=0 type=proc_start uuid="***402E" video_port=8000 vnc_scale=2
WARN[0248] Process end - stf_device_ios                  proc=stf_device_ios type=proc_end uuid="***402E"
INFO[0258] Process start - stf_device_ios                binary=/usr/local/opt/node@14/bin/node clickHeight=896 clickScale=1000 clickWidth=414 client_ip=192.168.1.11 device_name="Jitu iPhone Max" frame_server="ws://192.168.1.11:8000/echo" node_port=9240 proc=stf_device_ios server_host=127.0.0.1 server_ip=127.0.0.1 stream_height=0 stream_width=0 type=proc_start uuid="***402E" video_port=8000 vnc_scale=2
WARN[0258] Process end - stf_device_ios                  proc=stf_device_ios type=proc_end uuid="***402E"
INFO[0268] Process start - stf_device_ios                binary=/usr/local/opt/node@14/bin/node clickHeight=896 clickScale=1000 clickWidth=414 client_ip=192.168.1.11 device_name="Jitu iPhone Max" frame_server="ws://192.168.1.11:8000/echo" node_port=9240 proc=stf_device_ios server_host=127.0.0.1 server_ip=127.0.0.1 stream_height=0 stream_width=0 type=proc_start uuid="***402E" video_port=8000 vnc_scale=2
WARN[0269] Process end - stf_device_ios                  proc=stf_device_ios type=proc_end uuid="***402E"
INFO[0279] Process start - stf_device_ios                binary=/usr/local/opt/node@14/bin/node clickHeight=896 clickScale=1000 clickWidth=414 client_ip=192.168.1.11 device_name="Jitu iPhone Max" frame_server="ws://192.168.1.11:8000/echo" node_port=9240 proc=stf_device_ios server_host=127.0.0.1 server_ip=127.0.0.1 stream_height=0 stream_width=0 type=proc_start uuid="***402E" video_port=8000 vnc_scale=2
WARN[0279] Process end - stf_device_ios                  proc=stf_device_ios type=proc_end uuid="***402E"
INFO[0289] Process start - stf_device_ios                binary=/usr/local/opt/node@14/bin/node clickHeight=896 clickScale=1000 clickWidth=414 client_ip=192.168.1.11 device_name="Jitu iPhone Max" frame_server="ws://192.168.1.11:8000/echo" node_port=9240 proc=stf_device_ios server_host=127.0.0.1 server_ip=127.0.0.1 stream_height=0 stream_width=0 type=proc_start uuid="***402E" video_port=8000 vnc_scale=2
WARN[0290] Process end - stf_device_ios                  proc=stf_device_ios type=proc_end uuid="***402E"
INFO[0300] Process start - stf_device_ios                binary=/usr/local/opt/node@14/bin/node clickHeight=896 clickScale=1000 clickWidth=414 client_ip=192.168.1.11 device_name="Jitu iPhone Max" frame_server="ws://192.168.1.11:8000/echo" node_port=9240 proc=stf_device_ios server_host=127.0.0.1 server_ip=127.0.0.1 stream_height=0 stream_width=0 type=proc_start uuid="***402E" video_port=8000 vnc_scale=2
WARN[0301] Process end - stf_device_ios                  proc=stf_device_ios type=proc_end uuid="***402E"
INFO[0311] Process start - stf_device_ios                binary=/usr/local/opt/node@14/bin/node clickHeight=896 clickScale=1000 clickWidth=414 client_ip=192.168.1.11 device_name="Jitu iPhone Max" frame_server="ws://192.168.1.11:8000/echo" node_port=9240 proc=stf_device_ios server_host=127.0.0.1 server_ip=127.0.0.1 stream_height=0 stream_width=0 type=proc_start uuid="***402E" video_port=8000 vnc_scale=2
WARN[0312] Process end - stf_device_ios                  proc=stf_device_ios type=proc_end uuid="***402E"
INFO[0322] Process start - stf_device_ios                binary=/usr/local/opt/node@14/bin/node clickHeight=896 clickScale=1000 clickWidth=414 client_ip=192.168.1.11 device_name="Jitu iPhone Max" frame_server="ws://192.168.1.11:8000/echo" node_port=9240 proc=stf_device_ios server_host=127.0.0.1 server_ip=127.0.0.1 stream_height=0 stream_width=0 type=proc_start uuid="***402E" video_port=8000 vnc_scale=2
WARN[0322] Process end - stf_device_ios                  proc=stf_device_ios type=proc_end uuid="***402E"
INFO[0332] Process start - stf_device_ios                binary=/usr/local/opt/node@14/bin/node clickHeight=896 clickScale=1000 clickWidth=414 client_ip=192.168.1.11 device_name="Jitu iPhone Max" frame_server="ws://192.168.1.11:8000/echo" node_port=9240 proc=stf_device_ios server_host=127.0.0.1 server_ip=127.0.0.1 stream_height=0 stream_width=0 type=proc_start uuid="***402E" video_port=8000 vnc_scale=2
WARN[0333] Process end - stf_device_ios                  proc=stf_device_ios type=proc_end uuid="***402E"
INFO[0343] Process start - stf_device_ios                binary=/usr/local/opt/node@14/bin/node clickHeight=896 clickScale=1000 clickWidth=414 client_ip=192.168.1.11 device_name="Jitu iPhone Max" frame_server="ws://192.168.1.11:8000/echo" node_port=9240 proc=stf_device_ios server_host=127.0.0.1 server_ip=127.0.0.1 stream_height=0 stream_width=0 type=proc_start uuid="***402E" video_port=8000 vnc_scale=2
WARN[0343] Process end - stf_device_ios                  proc=stf_device_ios type=proc_end uuid="***402E"
INFO[0353] Process start - stf_device_ios                binary=/usr/local/opt/node@14/bin/node clickHeight=896 clickScale=1000 clickWidth=414 client_ip=192.168.1.11 device_name="Jitu iPhone Max" frame_server="ws://192.168.1.11:8000/echo" node_port=9240 proc=stf_device_ios server_host=127.0.0.1 server_ip=127.0.0.1 stream_height=0 stream_width=0 type=proc_start uuid="***402E" video_port=8000 vnc_scale=2
WARN[0354] Process end - stf_device_ios                  proc=stf_device_ios type=proc_end uuid="***402E"
INFO[0364] Process start - stf_device_ios                binary=/usr/local/opt/node@14/bin/node clickHeight=896 clickScale=1000 clickWidth=414 client_ip=192.168.1.11 device_name="Jitu iPhone Max" frame_server="ws://192.168.1.11:8000/echo" node_port=9240 proc=stf_device_ios server_host=127.0.0.1 server_ip=127.0.0.1 stream_height=0 stream_width=0 type=proc_start uuid="***402E" video_port=8000 vnc_scale=2
WARN[0364] Process end - stf_device_ios                  proc=stf_device_ios type=proc_end uuid="***402E"
INFO[0374] Process start - stf_device_ios                binary=/usr/local/opt/node@14/bin/node clickHeight=896 clickScale=1000 clickWidth=414 client_ip=192.168.1.11 device_name="Jitu iPhone Max" frame_server="ws://192.168.1.11:8000/echo" node_port=9240 proc=stf_device_ios server_host=127.0.0.1 server_ip=127.0.0.1 stream_height=0 stream_width=0 type=proc_start uuid="***402E" video_port=8000 vnc_scale=2
WARN[0375] Process end - stf_device_ios                  proc=stf_device_ios type=proc_end uuid="***402E"
INFO[0385] Process start - stf_device_ios                binary=/usr/local/opt/node@14/bin/node clickHeight=896 clickScale=1000 clickWidth=414 client_ip=192.168.1.11 device_name="Jitu iPhone Max" frame_server="ws://192.168.1.11:8000/echo" node_port=9240 proc=stf_device_ios server_host=127.0.0.1 server_ip=127.0.0.1 stream_height=0 stream_width=0 type=proc_start uuid="***402E" video_port=8000 vnc_scale=2
WARN[0386] Process end - stf_device_ios                  proc=stf_device_ios type=proc_end uuid="***402E"
INFO[0396] Process start - stf_device_ios                binary=/usr/local/opt/node@14/bin/node clickHeight=896 clickScale=1000 clickWidth=414 client_ip=192.168.1.11 device_name="Jitu iPhone Max" frame_server="ws://192.168.1.11:8000/echo" node_port=9240 proc=stf_device_ios server_host=127.0.0.1 server_ip=127.0.0.1 stream_height=0 stream_width=0 type=proc_start uuid="***402E" video_port=8000 vnc_scale=2
WARN[0396] Process end - stf_device_ios                  proc=stf_device_ios type=proc_end uuid="***402E"
INFO[0406] Process start - stf_device_ios                binary=/usr/local/opt/node@14/bin/node clickHeight=896 clickScale=1000 clickWidth=414 client_ip=192.168.1.11 device_name="Jitu iPhone Max" frame_server="ws://192.168.1.11:8000/echo" node_port=9240 proc=stf_device_ios server_host=127.0.0.1 server_ip=127.0.0.1 stream_height=0 stream_width=0 type=proc_start uuid="***402E" video_port=8000 vnc_scale=2
WARN[0407] Process end - stf_device_ios                  proc=stf_device_ios type=proc_end uuid="***402E"
INFO[0417] Process start - stf_device_ios                binary=/usr/local/opt/node@14/bin/node clickHeight=896 clickScale=1000 clickWidth=414 client_ip=192.168.1.11 device_name="Jitu iPhone Max" frame_server="ws://192.168.1.11:8000/echo" node_port=9240 proc=stf_device_ios server_host=127.0.0.1 server_ip=127.0.0.1 stream_height=0 stream_width=0 type=proc_start uuid="***402E" video_port=8000 vnc_scale=2
WARN[0418] Process end - stf_device_ios                  proc=stf_device_ios type=proc_end uuid="***402E"
INFO[0428] Process start - stf_device_ios                binary=/usr/local/opt/node@14/bin/node clickHeight=896 clickScale=1000 clickWidth=414 client_ip=192.168.1.11 device_name="Jitu iPhone Max" frame_server="ws://192.168.1.11:8000/echo" node_port=9240 proc=stf_device_ios server_host=127.0.0.1 server_ip=127.0.0.1 stream_height=0 stream_width=0 type=proc_start uuid="***402E" video_port=8000 vnc_scale=2
WARN[0428] Process end - stf_device_ios                  proc=stf_device_ios type=proc_end uuid="***402E"
INFO[0438] Process start - stf_device_ios                binary=/usr/local/opt/node@14/bin/node clickHeight=896 clickScale=1000 clickWidth=414 client_ip=192.168.1.11 device_name="Jitu iPhone Max" frame_server="ws://192.168.1.11:8000/echo" node_port=9240 proc=stf_device_ios server_host=127.0.0.1 server_ip=127.0.0.1 stream_height=0 stream_width=0 type=proc_start uuid="***402E" video_port=8000 vnc_scale=2
WARN[0439] Process end - stf_device_ios                  proc=stf_device_ios type=proc_end uuid="***402E"
INFO[0449] Process start - stf_device_ios                binary=/usr/local/opt/node@14/bin/node clickHeight=896 clickScale=1000 clickWidth=414 client_ip=192.168.1.11 device_name="Jitu iPhone Max" frame_server="ws://192.168.1.11:8000/echo" node_port=9240 proc=stf_device_ios server_host=127.0.0.1 server_ip=127.0.0.1 stream_height=0 stream_width=0 type=proc_start uuid="***402E" video_port=8000 vnc_scale=2
WARN[0449] Process end - stf_device_ios                  proc=stf_device_ios type=proc_end uuid="***402E"
INFO[0459] Process start - stf_device_ios                binary=/usr/local/opt/node@14/bin/node clickHeight=896 clickScale=1000 clickWidth=414 client_ip=192.168.1.11 device_name="Jitu iPhone Max" frame_server="ws://192.168.1.11:8000/echo" node_port=9240 proc=stf_device_ios server_host=127.0.0.1 server_ip=127.0.0.1 stream_height=0 stream_width=0 type=proc_start uuid="***402E" video_port=8000 vnc_scale=2
WARN[0460] Process end - stf_device_ios                  proc=stf_device_ios type=proc_end uuid="***402E"

**Comment by Jitu1888** [comment]
<img width="1503" alt="image" src="https://github.com/dryark/stf_ios_support/assets/8565346/cf907e4a-d8dc-40c4-9989-74c1491f3c1b">
